### PR TITLE
*: LOAD DATA support GBK character set

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #40499

Problem Summary:

### What is changed and how it works?

use lightning's CharsetConvertor

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/loadremotetest/one_csv_test.go
Comment (D3Hunter) [inline executor/loadremotetest/one_csv_test.go:1]: test for invalid charset?

Comment (lance6716) [reply executor/loadremotetest/one_csv_test.go:1]: https://github.com/pingcap/tidb/pull/42644/commits/b9bea88ed20383b029f31d5f292c2dd21bb11286

**Pull request merged**